src/sentry/api/endpoints/project_user_reports.py

```diff
@@ -106,9 +115,14 @@ def post(self, request, project):
             report.event_user_id = euser.id
 
         try:
-            report.group = Group.objects.from_event_id(project, report.event_id)
-        except Group.DoesNotExist:
-            pass
+            event = Event.objects.filter(event_id=report.event_id).select_related('group')[0]
+            report.environment = event.get_environment()
+            report.group = event.group
+        except IndexError:
+            try:
+                report.group = Group.objects.from_event_id(project, report.event_id)
+            except Group.DoesNotExist:
+                pass
```

Comment:
This should probably be structured as:

```python
try:
    event = Event.objects.filter(event_id=report.event_id).select_related('group')[0]
except IndexError:
    try:
        report.group = Group.objects.from_event_id(project, report.event_id)
    except Group.DoesNotExist:
        pass
else:
    report.environment = event.get_environment()
    report.group = event.group
```

This would prevent any `IndexError` raised on the assignment of `report.{environment,group}` (possible on method invocation or property access) from accidentally being trapped and changing the control flow here in ways that you didn't intend when you wrote the code.

In general, it's a good rule of thumb to put as few statements/expressions in the `try` section as possible to avoid catching exceptions that you didn't actually intend to catch.